nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/serving/SuggestionOp.java

```diff
@@ -77,27 +80,40 @@
     @Context
     protected MarshallerRegistry registry;
 
-    @Param(name = "document", description = "A document", required = false)
-    protected DocumentModel documentModel;
+    @Param(name = "updatedDocument", description = "Document with changes done on client side before saving", required = false)
+    protected DocumentModel updatedDocument;
 
     @Param(name = "references", description = "Should the entity references be resolved?", required = false)
     protected boolean references = false;
 
+    @OperationMethod
+    public Blob run(DocumentRef docRef) {
+        DocumentModel docModel = coreSession.getDocument(docRef);
+        return run(docModel);
+    }
+
     @OperationMethod
     public Blob run(DocumentModel doc) {
+        if (updatedDocument != null) {
+            Arrays.asList(doc.getSchemas())
```

Comment:
please no nested loops:

```
Arrays.stream(doc.getSchemas())
                            .flatMap(s1 -> udoc.getProperties(s1).entrySet().stream())
                            .forEach(entry -> doc.setPropertyValue(entry.getKey(), (Serializable) entry.getValue()));
```

should do the job

Reply:
Personally, in this example, I think a standard for loop will be much clearer

Reply:
depends, if nested loops don't scare you, then yes :)

Reply:
Thanks for your feedback!
Not afraid by nested loops (only two in fact), but I can understand your point @andreinechaev about the usage of streams here. I'll perform the change